middleware/src/handlers/handlers.go

```diff
@@ -0,0 +1,87 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	//"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+	//"github.com/tidwall/gjson"
+	"github.com/digitalbitbox/bitbox-base/middleware/src"
+	"github.com/digitalbitbox/bitbox-base/middleware/util/logging"
+	"github.com/sirupsen/logrus"
+)
+
+type Middleware interface {
+	Start() <-chan *middleware.MiddlewareInfoStruct
+}
+
+// Handlers provides a web api
+type Handlers struct {
+	Router     *mux.Router
+	clients    map[*websocket.Conn]bool
+	upgrader   websocket.Upgrader
+	log        *logrus.Entry
+	middleware Middleware
+	//TODO: In future this event should have a generic interface (thus only containing raw json)
+	middlewareEvents <-chan *middleware.MiddlewareInfoStruct
+}
+
+func NewHandlers(middlewareInstance Middleware) *Handlers {
+	router := mux.NewRouter()
+
+	handlers := &Handlers{
+		middleware: middlewareInstance,
+		Router:     router,
+		clients:    make(map[*websocket.Conn]bool),
+		upgrader: websocket.Upgrader{
```

Comment:
How about a brief comment explaining what this `upgrader` does?

Reply:
Comment added.

Reply:
Ah, I noticed you commented the field on L24 (`//upgrader takes an http request and upgrades the connection with its origin to websocket`). That's also useful, but I was referring to the `websocket.Upgrader{}` defined on this line, which always does `return true` in its `CheckOrigin` function.

Could we describe what we achieve by doing this?

Reply:
This is indeed more of a placeholder, but upon reading the documentation now, I see that this overrides the default origin checks. I removed it and added a comment that in future the Upgrader should do the origin check to ensure that the client is authenticated.